GVFS/GVFS.Platform.POSIX/POSIXPlatform.cs

```diff
@@ -124,6 +124,30 @@ public override bool TryGetGVFSEnlistmentRoot(string directory, out string enlis
             return POSIXPlatform.TryGetGVFSEnlistmentRootImplementation(directory, out enlistmentRoot, out errorMessage);
         }
 
+        public override bool TryGetDefaultLocalCacheRoot(string enlistmentRoot, out string localCacheRoot, out string localCacheRootError)
+        {
+            string homeDirectory = Environment.GetEnvironmentVariable("HOME");

```

Comment:
`HOME` seems like a really generic name for an environment variable should it be more specific?

Reply:
`HOME` is the environment variable that stores the location of the home directory on POSIX systems.  According to [this post](https://unix.stackexchange.com/questions/123858/is-the-home-environment-variable-always-set-on-a-linux-system) it will always be set.

Do you think a comment here would be helpful?

Reply:
One other detail, when you're using the terminal [~ is expanded to $HOME by Bash](https://www.gnu.org/software/bash/manual/html_node/Tilde-Expansion.html) and so we can't use `~` in the path ourselves.

Reply:
Are there any cases when the user would want to keep the `HOME` environment variable and have something else for the default VFSForGit cache root or if a user wants to do that they have to specify the cache location when cloning?

Reply:
> Are there any cases when the user would want to keep the HOME environment variable and have something else for the default VFSForGit cache root or if a user wants to do that they have to specify the cache location when cloning?

With these changes if the user wants the cache somewhere other than `~/.gvfsCache` they'd need to specify it when cloning.

We could consider adding our own environment variable to control this (e.g. `VFS4G_CLONE_CACHE_LOCATION`) but that work hasn't been done yet.

@jrbriggs let me know if you'd like me to file an issue for that feature and/or make that change in this PR.

Reply:
Let's treat this PR as the changing of the default. If we get user feedback we can add our own environment variable to control the cache location.